plugins/modules/cloud/misc/proxmox_snap.py

```diff
@@ -109,15 +119,15 @@ def snapshot_create(self, vm, vmid, timeout, snapname, description, vmstate):
         else:
             taskid = self.snapshot(vm, vmid).post(snapname=snapname, description=description, vmstate=int(vmstate))
         while timeout:
-            if (self.proxmox_api.nodes(vm['node']).tasks(taskid).status.get()['status'] == 'stopped' and
-                    self.proxmox_api.nodes(vm['node']).tasks(taskid).status.get()['exitstatus'] == 'OK'):
+            status_data = self.proxmox_api.nodes(vm['node']).tasks(taskid).status.get()
+            if status_data['status'] == 'stopped' and status_data['exitstatus'] == 'OK':
                 return True
-            timeout -= 1
```

Comment:
You'll have to mention the changes to other code in the changelog fragment as well (or better, have two fragments in the same file, one for the new feature and one for the refactoring / timeout fix).